src/App.tsx

```diff
@@ -1,14 +1,60 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const selectTodo = (todoId: number) => {
+    setSelectedTodo(todos.find(todo => todo.id === todoId) || null);
+  };
+
+  const closeModal = () => setSelectedTodo(null);
+
+  const filterTodos = () => {
+    const cleanQuery = query.trim().toLowerCase();
+
+    switch (filterType) {
+      case 'active':
+        return todos
+          .filter(todo => !todo.completed
+              && todo.title.toLowerCase().includes(cleanQuery));
+      case 'completed':
+        return todos
+          .filter(todo => todo.completed
+              && todo.title.toLowerCase().includes(cleanQuery));
+      default:
+        return todos
+          .filter(todo => todo.title.toLowerCase().includes(cleanQuery));
```

Comment:
feels too much of repeats
this seems to be valid - see if that fits your implementation
```suggestion
    return todos.filter(({ title, completed }) => (
      title.toLowerCase().includes(cleanQuery)
      && completed === (filterType === 'active')
    ))
```

Reply:
I actually thought along these lines, but decided to keep the switch, in case I'd want to extend this or make changes in the future. However I'll just simplify because it is like 1/3rd of the code